erigon-lib/kv/temporal/kv_temporal.go

```diff
@@ -179,65 +181,224 @@ func (db *DB) Close() {
 
 func (db *DB) OnFilesChange(f kv.OnFilesChange) { db.agg.OnFilesChange(f) }
 
-type Tx struct {
-	*mdbx.MdbxTx
+type tx struct {
 	db               *DB
 	aggtx            *state.AggregatorRoTx
 	resourcesToClose []kv.Closer
 	ctx              context.Context
+	mu               sync.RWMutex
```

Comment:
from mutex name must be clear which fields it protecting. (or at-least comments).